File: sdks/php/src/ErrorClassifier.php
<?php
namespace Satusehat\IntegrationSdk;
final class ErrorClassifier {public static function classify(int $s):array{if($s>=200&&$s<=299)return['success',false];return match($s){401=>['unauthorized',true],403=>['forbidden',false],404=>['not_found',false],409=>['conflict',false],422=>['validation_error',false],429=>['rate_limited',true],default=>$s>=500&&$s<=599?['server_error',true]:($s>=400&&$s<=499?['invalid_request',false]:['unexpected_http_status',false])};}}

File: sdks/ruby/lib/satusehat_sdk/error_classifier.rb
module Satusehat
    module IntegrationSdk
        module ErrorClassifier
            module_function

            def classify(status)
                status = Integer(status)
                return ["success", false] if status >= 200 && status <= 299

                case status
                when 401 then ["unauthorized", true]
                when 403 then ["forbidden", false]
                when 404 then ["not_found", false]
                when 409 then ["conflict", false]
                when 422 then ["validation_error", false]
                when 429 then ["rate_limited", true]
                when 500..599 then ["server_error", true]
                when 400..499 then ["invalid_request", false]
                else ["unexpected_http_status", false]
                end
            end
        end
    end
end
